src/rule_definition_tools.jl

```diff
@@ -341,19 +343,22 @@ function tuple_expression(primal_sig_parts)
     end
 end
 
```

Comment:
I wonder if we wouldn't be better off defining:
```
"changes `f(x,y)` into `f(x,y; kwargs....)`"
function _with_kwargs_expr(call_expr::Expr)
    kw_expr = Expr(:parameters, :(kwargs...))
    reutrn Expr(call_expr.head, call_expr.args[1], kw_expr, call_expr.args[2:end]...)
end
```
and then calling that from `_nondiff_rrule_expr` on `primal_invoke`
Then we could avoid generating 2 expressions here and passing them down.